From: Anish Bhatt <anish@chelsio.com>
Date: Tue, 14 Oct 2014 20:07:22 -0700
Subject: [PATCH 215/223] cxgb4 : Fix build failure in cxgb4 when ipv6 is
 disabled/not in-built
Patch-mainline: v3.18-rc1
Git-commit: 1bb60376cda108306818365b186450f154ede5f2
References: bsc#909577 FATE#317550

cxgb4 ipv6 does not guard against ipv6 being disabled, or the standard
ipv6 module vs inbuilt tri-state issue. This was fixed for cxgb4i & iw_cxgb4
but missed for cxgb4.

Signed-off-by: Anish Bhatt <anish@chelsio.com>
Signed-off-by: David S. Miller <davem@davemloft.net>

[dchang: refreshed, replaced IS_ENABLED(CONFIG_IPV6) with
defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)]
Acked-by: David Chang <dchang@suse.com>
---
 drivers/net/ethernet/chelsio/Kconfig            |    2 +-
 drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c |    8 ++++++++
 2 files changed, 9 insertions(+), 1 deletion(-)

--- a/drivers/net/ethernet/chelsio/Kconfig
+++ b/drivers/net/ethernet/chelsio/Kconfig
@@ -67,7 +67,7 @@ config CHELSIO_T3
 
 config CHELSIO_T4
 	tristate "Chelsio Communications T4/T5 Ethernet support"
-	depends on PCI
+	depends on PCI && (IPV6 || IPV6=n)
 	select FW_LOADER
 	select MDIO
 	---help---
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -4296,6 +4296,7 @@ EXPORT_SYMBOL(cxgb4_unregister_uld);
  * success (true) if it belongs otherwise failure (false).
  * Called with rcu_read_lock() held.
  */
+#if defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)
 static bool cxgb4_netdev(const struct net_device *netdev)
 {
 	struct adapter *adap;
@@ -4449,6 +4450,7 @@ static void update_clip(const struct ada
 	}
 	rcu_read_unlock();
 }
+#endif /* IS_ENABLED(CONFIG_IPV6) */
 
 /**
  *	cxgb_up - enable the adapter
@@ -4495,7 +4497,9 @@ static int cxgb_up(struct adapter *adap)
 	t4_intr_enable(adap);
 	adap->flags |= FULL_INIT_DONE;
 	notify_ulds(adap, CXGB4_STATE_UP);
+#if defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)
 	update_clip(adap);
+#endif
  out:
 	return err;
  irq_err:
@@ -6757,14 +6761,18 @@ static int __init cxgb4_init_module(void
 	if (ret < 0)
 		debugfs_remove(cxgb4_debugfs_root);
 
+#if defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)
 	register_inet6addr_notifier(&cxgb4_inet6addr_notifier);
+#endif
 
 	return ret;
 }
 
 static void __exit cxgb4_cleanup_module(void)
 {
+#if defined(CONFIG_IPV6) || defined(CONFIG_IPV6_MODULE)
 	unregister_inet6addr_notifier(&cxgb4_inet6addr_notifier);
+#endif
 	pci_unregister_driver(&cxgb4_driver);
 	debugfs_remove(cxgb4_debugfs_root);  /* NULL ok */
 }
